File: algorithms/SearchInsertPosition/SearchInsertPosition.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} target
    # @return {integer}
    def searchInsert(self, nums, target):
        low, high = 0, len(nums) - 1
        while low <= high:
            mid = (low + high) / 2
            if target < nums[mid]:
                high = mid - 1
            elif target > nums[mid]:
                low = mid + 1
            else:
                return mid
        return low

File: algorithms/SearchInsertPosition/SearchInsertPosition.c
int searchInsert(int* nums, int numsSize, int target) {
    int low = 0;
    int high = numsSize -1;
    int mid;
    while (low <= high) {
        mid = (low + high) / 2;
        if (target < nums[mid])
            high = mid - 1;
        else if (target > nums[mid])
            low = mid + 1;
        else
            return mid;
    }
    return low;
}